tests/spi/remote-secret.go

```diff
@@ -0,0 +1,126 @@
+package spi
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-appstudio/e2e-tests/pkg/framework"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	"github.com/redhat-appstudio/remote-secret/api/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+/*
+ * Component: spi
+ * Description: SVPI-541 - Basic remote secret functionalities
+ */
+
+var _ = framework.SPISuiteDescribe(Label("spi-suite", "remote-secret"), func() {
+
+	defer GinkgoRecover()
+
+	var fw *framework.Framework
+	var err error
+	var namespace string
+	var remoteSecret *v1beta1.RemoteSecret
+	targetNamespace1 := "spi-test-target1"
+	targetNamespace2 := "spi-test-target2"
+	remoteSecretName := "test-remote-secret"
+	targetSecretName1 := ""
+	targetSecretName2 := ""
+
+	Describe("SVPI-541 - Basic remote secret functionalities", Ordered, func() {
+		BeforeAll(func() {
+			// Initialize the tests controllers
+			fw, err = framework.NewFramework(utils.GetGeneratedNamespace("spi-demos"))
+			Expect(err).NotTo(HaveOccurred())
+			namespace = fw.UserNamespace
+			Expect(namespace).NotTo(BeEmpty())
+
+			_, err = fw.AsKubeAdmin.CommonController.CreateTestNamespace(targetNamespace1)
+			Expect(err).NotTo(HaveOccurred(), "Error when creating %s namespace: %v", targetNamespace1, err)
+
+			_, err = fw.AsKubeAdmin.CommonController.CreateTestNamespace(targetNamespace2)
+			Expect(err).NotTo(HaveOccurred(), "Error when creating %s namespace: %v", targetNamespace2, err)
+		})
+
+		AfterAll(func() {
+			if !CurrentSpecReport().Failed() {
+				Expect(fw.AsKubeAdmin.SPIController.DeleteAllRemoteSecretsInASpecificNamespace(namespace)).To(Succeed())
+				Expect(fw.AsKubeAdmin.CommonController.DeleteAllSecretsInASpecificNamespace(namespace)).To(Succeed())
+				Expect(fw.AsKubeAdmin.CommonController.DeleteAllSecretsInASpecificNamespace(targetNamespace1)).To(Succeed())
+				Expect(fw.AsKubeAdmin.CommonController.DeleteAllSecretsInASpecificNamespace(targetNamespace2)).To(Succeed())
+			}
+		})
+
+		It("creates RemoteSecret with previously created namespaces as targets", func() {
+			remoteSecret, err = fw.AsKubeDeveloper.SPIController.CreateRemoteSecret(remoteSecretName, namespace, []string{targetNamespace1, targetNamespace2})
+			Expect(err).NotTo(HaveOccurred())
+
+			remoteSecret, err = fw.AsKubeDeveloper.SPIController.GetRemoteSecret(remoteSecret.Name, namespace)
+			Expect(err).NotTo(HaveOccurred())
+
+			dataObtained := meta.IsStatusConditionFalse(remoteSecret.Status.Conditions, "DataObtained")
+			Expect(dataObtained).To(BeTrue())
```

Comment:
I would rename this variable to `dataNotObtained` or something similar because `Expect(dataObtained).To(BeTrue())` invokes in me that this check ensures that the data are obtained when in fact, it checks that the data are not obtained because of `meta.IsStatusConditionFalse` .

Reply:
I agree. Thanks!!